src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1406,6 +1408,38 @@ VideoDecoder::~VideoDecoder() {
   }
 }
 
+std::tuple<int, int> getHeightAndWidthFromOptionsOrMetadata(
+    const VideoDecoder::VideoStreamDecoderOptions& options,
+    const VideoDecoder::StreamMetadata& metadata) {
+  return std::make_tuple(
+      options.height.value_or(*metadata.height),
+      options.width.value_or(*metadata.width));
+}
+
+std::tuple<int, int> getHeightAndWidthFromOptionsOrAVFrame(
+    const VideoDecoder::VideoStreamDecoderOptions& options,
+    AVFrame* avFrame) {
+  return std::make_tuple(
+      options.height.value_or(avFrame->height),
+      options.width.value_or(avFrame->width));
+}
+
```

Comment:
I'd actually prefer that we just call both of these functions `getHeightAndWidth` and allow C++ function overloading to determine which one to call. In C++, because the types of parameters are formally a part of the function signature, it's less common to encode the type of one of the parameters in the name. This, however, is a question of style, and I know @ahmadsharif1 may feel differently.